### PR TITLE
Add related links widget to deposit form

### LINKED ISSUE #218: Add links to related content to the work edit page
See [mockups](https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420954309)

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #218

### CODE REVIEW

File: app/views/works/show.html.erb [hunk 1]
```diff
@@ -27,4 +27,10 @@
   <% @work.related_works.each do |k| %>
     citation: <%= k.citation %>
   <% end %>
+
+  related_links = <%= @work.related_links.size %>
+  <% @work.related_links.each do |k| %>
```
Comment (mjgiarlo) [inline app/views/works/show.html.erb:32]: Rename `k` to `link` or some such?

File: app/controllers/works_controller.rb [hunk 2]
```diff
@@ -13,11 +13,11 @@ def new
 
     work = Work.new(work_type: params[:work_type],
                     subtype: params[:subtype],
-                    collection: collection,
-                    contributors: [Contributor.new])
+                    collection: collection)
     authorize! work
 
     @form = WorkForm.new(work)
+    @form.prepopulate!
```
Comment (mjgiarlo) [inline app/controllers/works_controller.rb:20]: Nice!

File: app/components/works/related_link_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      <%= render Works::RelatedLinkRowComponent.new(form: link_form) %>
+    <% end %>
+  </template>
+
+  <%= form.fields_for :related_links do |link_form| %>
+    <%= render Works::RelatedLinkRowComponent.new(form: link_form) %>
+  <% end %>
+
+  <div data-target="nested-form.add_item">
+    <%= button_tag '+ Add another link', class: "btn btn-outline-primary", data: { action: "nested-form#add_association" } %>
```
Comment (mjgiarlo) [inline app/components/works/related_link_component.html.erb:17]: Can you change the action to `nested-form#addAssociation` ? refs #344

File: app/components/works/related_link_row_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <%= form.label :link_title, 'Link title', class: 'form-label col-md-2' %>
+
+    <div class="col-md-9">
+      <%= form.hidden_field :_destroy %>
+      <%= form.text_field :link_title, class: 'form-control' %>
+    </div>
+
+    <div class="col-md-1">
+      <%= button_tag class: 'btn btn-sm float-right', aria: { label: 'Remove' },
+          data: { action: "click->nested-form#remove_association" } do %>
```
Comment (mjgiarlo) [inline app/components/works/related_link_row_component.html.erb:12]: Same as above

File: app/components/works/related_link_component.html.erb (hunk 3 above)
Comment (jcoyne) [reply app/components/works/related_link_component.html.erb]: No problem

File: app/components/works/related_link_row_component.html.erb (hunk 4 above)
Comment (jcoyne) [reply app/components/works/related_link_row_component.html.erb]: done.

**Review by mjgiarlo — APPROVED** [review]
Merge when :green_heart:

**Pull request merged**